### PR TITLE
feat: Add support for Action Item Categories

### LINKED ISSUE #1462: Admin - Add support for Action Items
**Is your feature request related to a problem? Please describe.**
Adding support for Action Items in talawa-admin.

**Describe the solution you'd like**
1. A screen for action item categories on the org leftDrawer, through which we could create, edit and view all the categories.
2. A screen for action items on the org leftDrawer, for similar crud functionalities. We could also add custom filter and search functionality for getting action items for a single category, assigned to a certain user, etc.
3. A screen on the event dashboard for action items associated with that event (as was previously the case for event projects). Here also, aside from the basic crud, we could add filter search functionality like the one described above.

**Additional context**
It's a companion issue for [this](https://github.com/PalisadoesFoundation/talawa-api/issues/1595) one in the api.

### PR DESCRIPTION
**Pull request by meetulr**
**What kind of change does this PR introduce?**
It adds support for Action Item Category feature

**Issue Number:**
Fixes #1462

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**
For large screens:

https://github.com/PalisadoesFoundation/talawa-admin/assets/55585268/01b7e4b7-d2f6-4082-a5ff-9899b478ad27

For small screens:

https://github.com/PalisadoesFoundation/talawa-admin/assets/55585268/fe9a9b20-3039-402d-a56c-3a5d01721065

**Summary**
This PR introduces the following changes:
1. Adding CRU functionalities for Action Item Categories.
2. Design adjustments for the "orgSettings" screen by incorporating tabs that cater to organization-specific features. At present, the interface displays “General” and “Action Item Categories” tabs. This design allows for the seamless addition of more such features in the future.

**Does this PR introduce a breaking change?**
No

**Additional Context**
This is the first PR in the series for this issue, as discussed [here](https://github.com/PalisadoesFoundation/talawa-admin/issues/1462#issuecomment-1913788145).

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@meetulr Please fix the failed test.

**Comment by Cioppolo14** [comment]
@meetulr Why wasn't the schema file updated automatically in the original PR? It seems like now other PRs will are showing the schema file as changed because of this missing update...is that possible? Can you provide me more insight?

**Comment by meetulr** [comment]
> @meetulr Why wasn't the schema file updated automatically in the original PR? It seems like now other PRs will are showing the schema file as changed because of this missing update...is that possible? Can you provide me more insight?

Actually, at the time I was working on that [issue](https://github.com/PalisadoesFoundation/talawa-api/issues/1595), the pre-commit hook was running linting and formatting scripts on the entire codebase (there was an issue at the time to fix this behaviour).
So I modified the scripts and the hook to be run only on my staged files. I commented out the old scripts in the hook file and used my new scripts, and it seems that there I'd accidently commented out the "gen:schema" script too.

You're right, other PR's in api will show that file as modified.
Though it won't cause any breaking change, and it's a minor fix, could safely be merged with those PRs.
Still... I could raise a PR in the api to fix it.

I apologize for my oversight.

**Comment by Cioppolo14** [comment]
@meetulr Yes, can you create an issue, and I'll assign it to you, and you could fix it quickly? Thank you!

**Comment by meetulr** [comment]
> @meetulr Yes, can you create an issue, and I'll assign it to you, and you could fix it quickly? Thank you!

Yes, I've kept everything ready, I'd just have to open a PR.
I'll create an issue in the api for this.

**Comment by meetulr** [comment]
> @palisadoes @Cioppolo14 the test is failing because of two reasons:
> 
> 1. I forgot to update the "schema.graphql" file in my api [PR](https://github.com/PalisadoesFoundation/talawa-api/pull/1648) for action items😅. So the queries and mutations I made are not found in the api's "schema.graphql", and the introspection fails.
> 2. There are some discrepancies with "UpdateUserInput" schema which is causing this test to fail on other PRs too.
> 
> The first one would be fixed with any new merge into api's develop branch (as _schema.graphql_ would auto generate and include the types that are missing here). Second would require some modifications in the codebase (could be an issue in itself).
> 
> Other than that, this PR is ready for review.

Ok... now the test is failing only because of the second reason I mentioned above (more details [here](https://github.com/PalisadoesFoundation/talawa-admin/pull/1524#issuecomment-1920478554)).

**Comment by palisadoes** [comment]
@rishav-jha-mech @noman2002 @palisadoes @beingnoble03 please review

File: src/GraphQl/Mutations/ActionItemCategoryMutations.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import gql from 'graphql-tag';
```
Comment (rishav-jha-mech) [inline src/GraphQl/Mutations/ActionItemCategoryMutations.ts:1]: Why create a separate file for these mutations ?

File: src/GraphQl/Queries/ActionItemCategoryQueries.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import gql from 'graphql-tag';
```
Comment (rishav-jha-mech) [inline src/GraphQl/Queries/ActionItemCategoryQueries.ts:1]: Why have you created a separate file just for this query ?

File: src/GraphQl/Mutations/ActionItemCategoryMutations.ts (hunk 1 above)
Comment (meetulr) [reply src/GraphQl/Mutations/ActionItemCategoryMutations.ts:1]: I saw some separate files for some mutations and queries, and I thought we were moving towards more modular code, that's why😅. Shall I move them back into the main file?

File: src/GraphQl/Queries/ActionItemCategoryQueries.ts (hunk 2 above)
Comment (meetulr) [reply src/GraphQl/Queries/ActionItemCategoryQueries.ts:1]: Same as above...

File: src/screens/OrgSettings/OrgSettings.tsx [hunk 3]
```diff
@@ -1,68 +1,151 @@
...
   });
 
+  const orgSettings: SettingType[] = ['general', 'actionItemCategories'];
+
+  const [orgSetting, setOrgSetting] = useState<SettingType>('general');
+
   document.title = t('title');
   const orgId = window.location.href.split('=')[1];
 
+  const dropDirection: DropDirection = 'down';
```
Comment (rishav-jha-mech) [inline src/screens/OrgSettings/OrgSettings.tsx:29]: You don't need create a variable for this

**Comment by meetulr** [comment]
Github action bot is making the "Update Documentation" commit automatically.

**Comment by palisadoes** [comment]
I updated it. Please try again

**Comment by meetulr** [comment]
A test is failing in a file not related to this PR.

**Comment by meetulr** [comment]
@rishav-jha-mech I've made the changes, let me know if anything else is required.
Thank you!

**Comment by palisadoes** [comment]
1. Please fix.
2. Our branch protection rules need to be tightened. I'm working on it.

**Comment by palisadoes** [comment]
1. One approach could be to submit a working version of the files.
2. The issue seems to be occurring because PRs may have been overwriting the base branches in other PRs

**Review by noman2002 — CHANGES_REQUESTED** [review]
Please fix the failing tests.

**Comment by meetulr** [comment]
@palisadoes the single test that is failing here is for a file not related to this PR. It's causing the test to fail on other PRs too.

Including that file in this PR might cause unnecessary confusion as to why it's been modified in a completely unrelated issue.

I have fixed it locally, it's a quick fix, I could open a PR for that seperately so that it can be merged quickly, and then that test won't fail on other PRs too.

Pleses let me know what to do.

**Comment by palisadoes** [comment]
OK just create a PR for that file against the issue for this PR

**Comment by meetulr** [comment]
Ok! the tests are passing now.

**Review by palisadoes — APPROVED** [review]

**Comment by palisadoes** [comment]
@meetulr 

This needs to be reverted. You seem to have missed an edge case in your tests. They passed with the PR.

- https://github.com/PalisadoesFoundation/talawa-admin/actions/runs/7863549480/job/21454284769#step:6:1127

**Comment by palisadoes** [comment]
Please resubmit your PR

**Pull request merged**